### ISSUE TITLE
bind-service's flag binding-name version check isn't conditional on using the flag

### ISSUE THREAD

**Issue by cholick** [cloudfoundry/cli#1359]
## Command  

`cf bind-service my-service my-app`

## What occurred

Error message
```
Option '--name' requires CF API version 2.99.0 or higher. Your target is 2.98.0.
```

## What you expected to occur

We're not using the `--name` option (side note: this error message is dated, the flag is now named `--binding-name`). I would expect the version check 
https://github.com/cloudfoundry/cli/blob/master/command/v2/bind_service_command.go#L57-L60 to be conditional on actually using the flag.

## CLI Version  

v6.36.0

## CC API Endpoint Version  

Older than 2.99.0

## Any other relevant information  

We bundle the `cf` binary in a wide variety of automation, and this feature could be made backward compatible by only surfacing an error in the case where the command actually uses the flag. Our alternative is pinning to an older version, which introduces its own set of problems.

**Comment by XenoPhex** [comment]
oof FIXING!!!!

**Comment by cholick** [comment]
Thanks for picking this up and working on it so quickly @XenoPhex! Appreciate it.

**Comment by XenoPhex** [comment]
Fixed in `cf 6.36.1`
